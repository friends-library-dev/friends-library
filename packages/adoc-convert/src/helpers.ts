export function makeReduceWrapper(
  before: string,
  after: string,
): (acc: string[], str: string, index: number, array: string[]) => string[] {
  return (acc = [], str, index, array) => {
    index === 0 && acc.unshift(before);
    acc.push(str);
    index === array.length - 1 && acc.push(after);
    return acc;
  };
}
