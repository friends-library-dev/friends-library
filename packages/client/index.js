require('@babel/register');
const path = require('path');
require('dotenv').config({ path: path.join(__dirname, '..', '..', '.env') });
const cloud = require('./src/cloud');

module.exports = {
  cloud,
};
