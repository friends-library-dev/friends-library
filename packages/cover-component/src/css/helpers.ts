import cx from 'classnames';
import { CoverProps, PrintSize, Css } from '@friends-library/types';
import { sizes as bookSizes } from '@friends-library/lulu';

export interface DocDims {
  width: number;
  height: number;
  pdfSpineWidth: number;
  threeDSpineWidth: number;
}

export function docDims(size: PrintSize, pages: number, scaler?: number): DocDims {
  const { width, height } = bookSizes[size].dims;
  const spinePad = 0.06;
  const pagesPerInch = 444;
  const threeDSpineWidth = spinePad + pages / pagesPerInch;
  const pdfSpineWidth = pages < 32 ? 0 : threeDSpineWidth;

  return {
    width,
    height,
    pdfSpineWidth,
    threeDSpineWidth,
  };
}

export function wrapClasses(
  {
    edition,
    lang,
    size,
    scope,
    scaler,
  }: Pick<CoverProps, 'edition' | 'lang' | 'size' | 'scope' | 'scaler'>,
  customClasses?: string | string[] | Record<string, boolean>,
): string {
  const scale = typeof scaler === 'number' ? scaler : 1;
  return cx(
    'Cover',
    `Edition--${edition}`,
    `Lang--${lang}`,
    `trim--${size}`,
    scope ? `Cover--scope-${scope}` : false,
    scale <= 0.5 ? 'Cover--scale-s' : false,
    scale <= 0.35 ? 'Cover--scale-xs' : false,
    customClasses,
  );
}

export function scaleCssInches(css: Css, scaler?: number): Css {
  if (typeof scaler !== 'number') {
    return css;
  }

  const pattern = /(\d*(?:\.\d+)?)in(?! +{)(;| |\))/g;

  return css.replace(pattern, (full, inches, after) => {
    return `${Number(inches) * scaler}in${after}`;
  });
}

export function scopeCss(css: Css, scope?: string): Css {
  if (!scope) {
    return css;
  }

  return css.replace(/\.Cover(?=\.| |,)/gm, `.Cover--scope-${scope}`);
}

export function dynamifyCss(css: Css, scope?: string, scaler?: number): Css {
  return scopeCss(scaleCssInches(css, scaler), scope);
}

/**
 * An identity pass-through tagged template literal function
 * just so I can get syntax highlighting etc. from vscode
 */
export function css(strings: any, ...values: any[]): string {
  let str = '';
  strings.forEach((string: string, i: number) => {
    str += string + (values[i] || '');
  });
  return str;
}
