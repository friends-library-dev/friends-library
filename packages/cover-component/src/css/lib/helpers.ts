import cx from 'classnames';
import { CoverProps } from '@friends-library/types';

/**
 * An identity pass-through tagged template literal function
 * just so I can get syntax highlighting etc. from vscode
 */
export function css(strings: any, ...values: any[]): string {
  let str = '';
  strings.forEach((string: string, i: number) => {
    str += string + (values[i] || '');
  });
  return str;
}

export function wrapClasses(
  { edition, lang, size }: CoverProps,
  customClasses?: string | string[] | Record<string, boolean>,
): string {
  return cx(
    'Cover',
    `Edition--${edition}`,
    `Lang--${lang}`,
    `trim--${size}`,
    customClasses,
  );
}
