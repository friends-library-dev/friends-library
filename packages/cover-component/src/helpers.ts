import React from 'react';
import { Html } from '@friends-library/types';
import { quotify } from '@friends-library/adoc-utils';

export function overridable(
  key: string,
  fragments: Record<string, Html>,
  fallback: JSX.Element,
): JSX.Element {
  if (fragments[key] !== undefined) {
    return React.createElement(key === 'blurb' ? 'div' : fallback.type, {
      className: key,
      dangerouslySetInnerHTML: { __html: fragments[key] },
    });
  }
  return fallback;
}

export function initials(author: string): [string, string] {
  const [first, ...rest] = author.split(' ');
  return [first[0].toUpperCase(), rest[rest.length - 1][0].toUpperCase()];
}

export function prepareTitle(title: string, name: string): string {
  title = title.replace(/--/g, '–');
  title = title.replace(/ – Volumen? (?<number>(\d+|[IV]+))/, ', Vol.&nbsp;$<number>');
  return title.replace(name, name.replace(/ /g, '&nbsp;'));
}

export function formatBlurb(blurb: string): string {
  return quotify(blurb)
    .replace(/"`/g, '“')
    .replace(/`"/g, '”')
    .replace(/'`/g, '‘')
    .replace(/`'/g, '’')
    .replace(/--/g, '–');
}
