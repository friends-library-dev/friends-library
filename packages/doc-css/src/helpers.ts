import fs from 'fs';
import { Css, DocPrecursor } from '@friends-library/types';

export function joinCssFiles(paths: string[]): Css {
  return paths.map(toCss).join(`\n`);
}

export function replaceVars(css: Css, vars: Record<string, string>): Css {
  return css.replace(/var\((--[^)]+)\)/g, (_, varId) => {
    return vars[varId];
  });
}

export function toCss(relPath: string): Css {
  // going all the way up to root allows this to work for ncc-bundled actions
  // which are in the same directory depth, but a different location
  const ROOT = `${__dirname}/../../..`;
  return fs.readFileSync(`${ROOT}/packages/doc-css/src/css/${relPath}.css`).toString();
}

export function runningHead(dpc: DocPrecursor): string {
  const { sections, meta, config } = dpc;
  return sections.length === 1 ? meta.author.name : config.shortTitle || meta.title;
}
