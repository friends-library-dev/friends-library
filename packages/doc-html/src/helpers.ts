import { Lang, Html } from '@friends-library/types';

export const br7 = `<br class="m7"/>`;

export function ucfirst(lower: string): string {
  return lower.replace(/^\w/, c => c.toUpperCase());
}

const smallEn = `a|an|and|as|at|but|by|en|for|if|in|of|on|or|the|to|via`.split(`|`);
const smallEs = `a|un|una|el|la|los|las|y|e|o|con|de|del|al|por|si|en`.split(`|`);

export function capitalizeTitle(str: string, lang: Lang): string {
  const small = lang === `en` ? smallEn : smallEs;
  return str
    .split(` `)
    .map((word, index, parts) => {
      if (index === 0 || index === parts.length - 1) {
        return ucfirst(word);
      }
      return small.includes(word.toLowerCase()) ? word : ucfirst(word);
    })
    .join(` `);
}

export function trimTrailingPunctuation(str: string): string {
  return str.replace(/(?<!etc)[.,]$/, ``);
}

export function removeMobi7Tags(html: Html): Html {
  return html
    .replace(/ *<br class="m7" *\/>\n?/gim, ``)
    .replace(/ *<span class="m7">.+?<\/span>\n?/gim, ``);
}
