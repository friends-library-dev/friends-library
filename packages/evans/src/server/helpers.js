// @flow
import * as React from 'react';
import ReactDOM from 'react-dom/server';
import { readFileSync } from 'fs';
import { safeLoad } from 'js-yaml';
import friendFromJS from '../classes/map';
import { renderStaticOptimized } from 'glamor/server';
import { minify, stripDataSelectors, format, appCss } from '../lib/css';
import Friend from '../classes/Friend';

const { env: { NODE_ENV, PORT } } = process;


export function getFriend(slug: string): Friend {
  const path = `./node_modules/@friends-library/friends/src/en/${slug}.yml`;
  const file = readFileSync(path);
  const data: Object = safeLoad(file);
  return friendFromJS(data);
}


export const wrap = (Component: React.Element<*>): string => {
  const { html, css: glamorCss } = renderStaticOptimized(() => {
    return ReactDOM.renderToStaticMarkup(Component);
  });

  let markup = `<!doctype html>${html}`;
  const css = `${minify(appCss())}${stripDataSelectors(glamorCss)}`;
  markup = markup.replace('</head>', `<style>${format(css)}</style></head>`);

  if (NODE_ENV === 'development') {
    const bsUri = 'http://localhost:2223/browser-sync/browser-sync-client.js?v=2.23.3';
    markup = markup.replace(
      '</body>',
      `<script async src="${bsUri}"></script></body>`,
    );
  }

  return markup;
}
