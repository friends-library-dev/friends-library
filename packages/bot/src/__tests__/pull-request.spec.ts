import { cloud } from '@friends-library/client';
import kiteCheck from '../check/kite';
import lintCheck from '../check/lint';
import { prTestSetup } from './helpers';
import pullRequest from '../pull-request';

jest.mock('@friends-library/client');
jest.mock('@friends-library/asciidoc');
jest.mock('../check/kite');
jest.mock('../check/lint');

describe('pullRequest()', () => {
  let payload: any;
  let github: any;
  let context: any;

  beforeEach(() => {
    [context, github, payload] = prTestSetup();
  });

  it('does not do asciidoc checks for monorepo', () => {
    payload.repository.name = 'friends-library';
    pullRequest(context);
    expect(github.checks.create).not.toHaveBeenCalled();
    expect(kiteCheck).not.toHaveBeenCalled();
    expect(lintCheck).not.toHaveBeenCalled();
  });

  it('adds netlify comment for monorepo PRs', () => {
    payload.repository.name = 'friends-library';
    payload.action = 'opened';
    pullRequest(context);
    expect(github.issues.createComment).toHaveBeenCalledWith({
      number: 11,
      owner: 'friends-library-sandbox',
      repo: 'friends-library',
      body: expect.any(String),
    });
    expect(github.issues.createComment.mock.calls[0][0].body).toMatchSnapshot();
  });

  it('does not add netlify comment when monorepo PR not opened', () => {
    payload.repository.name = 'friends-library';
    payload.action = 'synchronize';
    pullRequest(context);
    expect(github.issues.createComment).not.toHaveBeenCalled();
  });

  it('requests modified files for PR', () => {
    pullRequest(context);
    expect(github.pullRequests.listFiles).toHaveBeenCalledWith({
      owner: 'friends-library-sandbox',
      repo: 'jane-doe',
      number: 11,
      per_page: 100,
    });
  });

  it('fetches file content for PR modified files', async () => {
    await pullRequest(context);
    await Promise.resolve();
    await Promise.resolve();

    expect(github.repos.getContents).toHaveBeenCalledWith({
      owner: 'friends-library-sandbox',
      repo: 'jane-doe',
      path: '01.adoc',
      ref: '2d306bb70578e6c019e3579c02d4f78f17bf915e',
    });
  });

  it('passes fetched files to lint and kite checks', async () => {
    const files = [{ path: '01.adoc', adoc: '== Ch 1' }];
    await pullRequest(context);
    expect(lintCheck).toHaveBeenCalledWith(expect.anything(), files);
    expect(kiteCheck).toHaveBeenCalledWith(expect.anything(), files);
  });

  it('deletes cloud PR preview files on PR close', () => {
    payload.action = 'closed';
    pullRequest(context);
    expect(cloud.rimraf).toHaveBeenCalledWith('pull-request/jane-doe/11');
  });
});
