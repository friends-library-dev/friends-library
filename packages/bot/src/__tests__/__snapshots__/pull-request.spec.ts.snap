// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pullRequest() adds netlify comment for monorepo PRs 1`] = `
"Deploy previews:

- [Website (English)](https://deploy-preview-11--en-evans.netlify.app)
- [Website (Spanish)](https://deploy-preview-11--es-evans.netlify.app)
- [Online Editor](https://deploy-preview-11--flp-editor.netlify.app)
- [Styleguide](https://deploy-preview-11--flp-styleguide.netlify.app)
- [Covers](https://deploy-preview-11--flp-covers.netlify.app)
- [Storybook](https://deploy-preview-11--flp-storybook.netlify.app)"
`;
